BaseApp/COD50014.TXT

```diff
@@ -0,0 +1,61 @@
+OBJECT Codeunit 50014 DotNet_CryptoStream
+{
+  OBJECT-PROPERTIES
+  {
+    Date=11/07/18;
+    Time=[ 9:09:28 AM];
+    Modified=Yes;
+    Version List=;
+  }
+  PROPERTIES
+  {
+    OnRun=BEGIN
+          END;
+
+  }
+  CODE
+  {
+    VAR
+      DotNetCryptoStream@1000000000 : DotNet "'mscorlib'.System.Security.Cryptography.CryptoStream";
+
+    [External]
+    PROCEDURE CryptoInStream@1000000005(IStream@1000000000 : InStream;VAR DotNet_ICryptoTransform@1000000001 : Codeunit 50013);
+    VAR
+      DotNetICryptoTransform@1000000004 : DotNet "'mscorlib'.System.Security.Cryptography.ICryptoTransform";
+      DotNetCryptoStreamMode@1000000003 : DotNet "'mscorlib'.System.Security.Cryptography.CryptoStreamMode";
+    BEGIN
+      DotNet_ICryptoTransform.GetICryptoTransform(DotNetICryptoTransform);
+      DotNetCryptoStream := DotNetCryptoStream.CryptoStream(IStream, DotNetICryptoTransform, DotNetCryptoStreamMode.Read);
```

Comment:
This should be a wrapper and included in the parameters of the constructor 😄

Reply:
Can you elaborate on this please?
Did you want the IStream/OStream wrapped into a DotNet_Stream?

Reply:
I meant [CryptoStreamMode](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.cryptostreammode?view=netframework-4.7.2) should be a wrapper on its own and passed as an argument to the constructor (not hardcoded).

Something like: 
`PROCEDURE CryptoInStream@1000000005(IStream@1000000000 : InStream;VAR DotNet_ICryptoTransform@1000000001 : Codeunit DotNet_ICryptoTransform;DotNet_CryptoStreamMode@1000000001 : Codeunit DotNet_CryptoStreamMode)`

Reply:
Ahh, I think I get it now.
You hardcoded the stream mode value because the first parameter is of type `InStream`.

Okay, so it was my bad.